src/main/scala/amba/axi4/Xbar.scala

```diff
@@ -148,38 +148,42 @@ class AXI4Xbar(
 
         // Block AW if we cannot record the W destination
         val allowAW = awFIFOMap(io_in(i).aw.bits.id)
-        val latched = RegInit(Bool(false)) // cut awIn(i).enq.valid from awready
+        val latched = RegInit(false.B) // cut awIn(i).enq.valid from awready
         in(i).aw.valid := io_in(i).aw.valid && (latched || awIn(i).io.enq.ready) && allowAW
         io_in(i).aw.ready := in(i).aw.ready && (latched || awIn(i).io.enq.ready) && allowAW
         awIn(i).io.enq.valid := io_in(i).aw.valid && !latched
-        when (awIn(i).io.enq.fire()) { latched := Bool(true) }
-        when (in(i).aw.fire()) { latched := Bool(false) }
+        when (awIn(i).io.enq.fire) { latched := true.B }
+        when (in(i).aw.fire) { latched := false.B }
 
         // Block W if we do not have an AW destination
         in(i).w.valid := io_in(i).w.valid && awIn(i).io.deq.valid // depends on awvalid (but not awready)
         io_in(i).w.ready := in(i).w.ready && awIn(i).io.deq.valid
         awIn(i).io.deq.ready := io_in(i).w.valid && io_in(i).w.bits.last && in(i).w.ready
+      } else {
+        awIn(i).io := DontCare // aw in queue is not used when outsize == 1
       }
     }
 
     // Transform output bundles
-    val out = Wire(Vec(io_out.size, AXI4Bundle(wide_bundle)))
+    val out = Wire(Vec(io_out.size, new AXI4Bundle(wide_bundle)))
     for (i <- 0 until out.size) {
       io_out(i) :<> out(i)
 
       if (io_in.size > 1) {
         // Block AW if we cannot record the W source
-        val latched = RegInit(Bool(false)) // cut awOut(i).enq.valid from awready
+        val latched = RegInit(false.B) // cut awOut(i).enq.valid from awready
         io_out(i).aw.valid := out(i).aw.valid && (latched || awOut(i).io.enq.ready)
         out(i).aw.ready := io_out(i).aw.ready && (latched || awOut(i).io.enq.ready)
         awOut(i).io.enq.valid := out(i).aw.valid && !latched
-        when (awOut(i).io.enq.fire()) { latched := Bool(true) }
-        when (out(i).aw.fire()) { latched := Bool(false) }
+        when (awOut(i).io.enq.fire) { latched := true.B }
+        when (out(i).aw.fire) { latched := false.B }
 
         // Block W if we do not have an AW source
         io_out(i).w.valid := out(i).w.valid && awOut(i).io.deq.valid // depends on awvalid (but not awready)
         out(i).w.ready := io_out(i).w.ready && awOut(i).io.deq.valid
         awOut(i).io.deq.ready := out(i).w.valid && out(i).w.bits.last && io_out(i).w.ready
+      } else {
+        awOut(i).io := DontCare // aw out queue is not used when outsize == 1
```

Comment:
I don't think the comment is complied with the code.

Reply:
Thanks, addressed with https://github.com/chipsalliance/rocket-chip/pull/3059/commits/dd3703cb88a91ae4713b61be2c5e37a83fbcc77f